geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/LiveKafkaConsumerFeatureSource.scala

```diff
@@ -114,19 +124,86 @@ class LiveKafkaConsumerFeatureSource(entry: ContentEntry,
     }, 0, 1, TimeUnit.SECONDS)
   }
 
-  override def run(): Unit = while (running.get) {
-    queue.take() match {
-      case update: CreateOrUpdate => featureCache.createOrUpdateFeature(update)
-      case del: Delete            => featureCache.removeFeature(del)
-      case clr: Clear             => featureCache.clear()
-      case m                      => throw new IllegalArgumentException(s"Unknown message: $m")
+  override def run(): Unit =
+    while (running.get) {
+      queue.take() match {
+        case update: CreateOrUpdate =>
+          fireEvent(KafkaFeatureEvent.changed(this, update.feature))
```

Comment:
@anthonyccri correctly noted that we ought to minimize object creation for the more common, no subscribed feature listeners.

I think the most sensible way to support exactly what you are describing would be to push the listeners down into the FeatureCache.

ContentFeatureSouce.addFeatureListeners is sadly final, so doing that would require a bit of hack.